project-base/app/tests/FrontendApiBundle/Functional/SpecialPrice/ProductPriceQueryTest.php

```diff
@@ -0,0 +1,482 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\FrontendApiBundle\Functional\SpecialPrice;
+
+use App\DataFixtures\Demo\ProductDataFixture;
+use DateTimeImmutable;
+use PHPUnit\Framework\Attributes\DataProvider;
+use Shopsys\FrameworkBundle\Component\Money\HiddenMoney;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+use Shopsys\FrameworkBundle\Model\Product\ProductTypeEnum;
+use Shopsys\FrontendApiBundle\Model\Resolver\Price\ProductPriceQuery;
+use Tests\App\Test\TransactionFunctionalTestCase;
+
+final class ProductPriceQueryTest extends TransactionFunctionalTestCase
```

Comment:
quite an unusual way of testing the API :thinking: I am not saying there is anything wrong with that, I am just curious, is there any particular reason why you do not use `GraphQlTestCase`?

Reply:
speed of execution and the necessity of a specific setup for each test scenario. It would require many updates of the real data or adjusting the datafixtures (add more variants, add more price lists, and find the combination where I can test all scenarios I was thinking of)